components/collator/src/elements.rs

```diff
@@ -142,16 +142,18 @@ const NO_CE_VALUE: u64 =
     ((NO_CE_PRIMARY as u64) << 32) | ((NO_CE_SECONDARY as u64) << 16) | (NO_CE_TERTIARY as u64); // 0x101000100
 
 // See ICU4C collation.h and https://www.unicode.org/reports/tr10/#Trailing_Weights
-const FFFD_PRIMARY: u32 = 0xFFFD0000; // U+FFFD
+const FFFD_PRIMARY: u32 = REPLACEMENT_CHARACTER as u32; // U+FFFD
```

Comment:
Issue: `REPLACEMENT_CHARACTER as u32` is `0x0000fffd` not `0xfffd0000` as this variable had before

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=7a1db88137c9cc82ed6bf4fe05660157

I'm actually not convinced that this is a positive change, because it makes the bit patterns less explicit.

Reply:
> Issue: `REPLACEMENT_CHARACTER as u32` is `0x0000fffd` not `0xfffd0000` as this variable had before

Good catch!
As the _PRIMARY suffix suggests, this is a primary *weight* (left-aligned), not a code point. And the weight may not always match the left-shifted code point, so resist the temptation to compute it as `REPLACEMENT_CHARACTER << 16`. Please just revert this line.